build.gradle

```diff
@@ -1757,6 +1757,7 @@ project(':tools') {
   archivesBaseName = "kafka-tools"
 
   dependencies {
+    implementation project(':core')
```

Comment:
Ticket https://issues.apache.org/jira/browse/KAFKA-14525 (the parent of https://issues.apache.org/jira/browse/KAFKA-14586) says 

> tools that don't require access to `core` classes and communicate via the kafka protocol (typically by using the client classes) should be moved to the `tools` module.

This addition contradicts that requirement.

As far as I see, `kafka.utils.CommandLineUtils` is the only dependency to `core`. Is that true? Would it be possible to move `kafka.utils.CommandLineUtils` to a different module? Or should we even get completely rid of that dependency by rewriting it in java and put it in a different module.

https://issues.apache.org/jira/browse/KAFKA-14576 should have a similar issue since also the console consumer should be moved to tools and it has a dependency to  `kafka.utils.CommandLineUtils`.

Reply:
Hello, hello, chipping in to say this is the same conversation we had with @fvaleri in https://github.com/apache/kafka/pull/13122. There should be a separate PR soon containing changes for CommandLineUtils (et al.) to unblock all JIRA tickets depending on it 😊

Reply:
Thanks @clolov i was about to type the same thing :D So this PR and I guess your PR is blocked until https://github.com/apache/kafka/pull/13122 is merged?

Reply:
Also, not sure if this comment from Ismael has any relevance here: https://github.com/apache/kafka/pull/13095#issuecomment-1376021193?

Reply:
Ah, great! Thanks for sharing! Good to know that people is aware!

> Also, not sure if this comment from Ismael has any relevance here: https://github.com/apache/kafka/pull/13095#issuecomment-1376021193?

I think there they talk about the dependencies of tools' test module not of the tools per se.

Reply:
Yes I'm working on https://issues.apache.org/jira/browse/KAFKA-14628 and really close to opening the PR.

Reply:
@cadonna yes, there is some confusion on which tools need to be migrated. For example, `DumpLogSegments` has multiple dependencies on `core` classes, so it should be excluded, but was added as a sub task. Checking with @mimaison how we want to precede with these. All the rest can be migrated.

Reply:
btw, @fvaleri , unrelated to this PR, but there seems to another core dependency from `ReplicaVerificationTools`. Plz check my comment on https://issues.apache.org/jira/browse/KAFKA-14583. I can file an issue and migrate that one as well if you can also confirm the same. Thanks

Reply:
hi @cadonna , The PR for moving `CommandLineUtils` has been [merged to trunk](https://github.com/apache/kafka/pull/13131), meaning I don't need to include this dependency anymore. I have updated the same.